client-extensions/liferay-object-collection-filter-extension/src/common/components/SearchResult.js

```diff
@@ -0,0 +1,50 @@
+import { Body, Cell, Head, Row, Table } from "@clayui/core";
+import { useEffect, useState } from "react";
+
+const excludeCode = ["externalReferenceCode", "id","actions", "creator", "status", "_key", "keywords"]
+
+const SearchResult = ({items}) => {
+	const [heads, setHeads] = useState([]);
+
+	useEffect(() => {
+		let columns = [];
+
+		if(items && items.length > 0) {
```

Comment:
could you move the logic outside useEffect 
const generateColumns = (data, excludeColumns = []) => {
your logic here
}
and 
useEffect(() => {
    const columns = generateColumns(items, excludeCode);
    setHeads(columns);
  }, [items, excludeCode]);